client/src/app/components/Button/Button.js

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import "./buttons.css";
+const classNames = require("classnames");
+
+class ButtonContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { countActive: 0, isMax: false };
+  }
+
+  componentWillUpdate(props) {
+    if (props.reset !== this.props.reset) {
+      if (!this.props.reset) {
+        this.setState(() => ({ isMax: false, countActive: 0 }));
+      }
+    }
+  }
+
+  checkMax(num) {
+    this.setState(
+      prevState => ({ countActive: prevState.countActive + num }),
+      () => {
+        if (this.state.countActive === this.props.maxAllowed) {
+          this.setState({ isMax: true });
+        } else {
+          this.setState({ isMax: false });
+        }
+      }
+    );
+  }
+
+  render() {
+    return (
+      <div className="buttons-container">
+        {this.props.buttons.map((button, index) => (
+          <Button
+            key={index}
+            name={button}
+            checkMax={(num, title) => this.checkMax(num)}
+            isMax={this.state.isMax}
+            onUpdateSelection={this.props.onUpdateSelection}
+            default={this.props.default ? this.props.default : ""}
+            reset={this.props.reset}
+          />
+        ))}
+      </div>
+    );
+  }
+}
+
+class Button extends React.Component {
```

Comment:
'Button' and 'ButtonContainer' are 2 completely different components. They should be separated in their own files. 

One component per file is best.